src/main/java/org/broadinstitute/hellbender/tools/walkers/rnaseq/GeneExpressionEvaluation.java

```diff
@@ -0,0 +1,667 @@
+import htsjdk.samtools.SAMTag;
+import htsjdk.samtools.SAMUtils;
+import htsjdk.samtools.TextCigarCodec;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+import htsjdk.samtools.util.OverlapDetector;
+import htsjdk.tribble.annotation.Strand;
+import htsjdk.tribble.gff.Gff3BaseData;
+import htsjdk.tribble.gff.Gff3Feature;
+import org.apache.commons.lang3.tuple.Pair;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import org.broadinstitute.hellbender.cmdline.StandardArgumentDefinitions;
+import org.broadinstitute.hellbender.cmdline.programgroups.CoverageAnalysisProgramGroup;
+import org.broadinstitute.hellbender.engine.*;
+import org.broadinstitute.hellbender.engine.filters.AlignmentAgreesWithHeaderReadFilter;
+import org.broadinstitute.hellbender.engine.filters.MappingQualityReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilter;
+import org.broadinstitute.hellbender.engine.filters.ReadFilterLibrary;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+
+import org.broadinstitute.hellbender.utils.IntervalUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+
+import org.broadinstitute.hellbender.utils.tsv.DataLine;
+import org.broadinstitute.hellbender.utils.tsv.TableColumnCollection;
+import org.broadinstitute.hellbender.utils.tsv.TableReader;
+import org.broadinstitute.hellbender.utils.tsv.TableWriter;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Evaluate gene expression from RNA-seq reads aligned to genome.
+ *
+ * <p>This tool evaluates gene expression from RNA-seq reads aligned to genome.  Features to evaluate expression over are defined in an input annotation file in gff3 fomat
+ * (https://github.com/The-Sequence-Ontology/Specifications/blob/master/gff3.md).  Output is a tsv listing sense and antisense expression for all stranded grouping features,
+ * and expression (labeled as sense) for all unstranded grouping features.
+ * </p>
+ * 
+ * <p>
+ *     <h3>Input</h3>
+ *     <ul>
+ *     <li>BAM file of RNA-seq reads</li>
+ *     <li>Gff3 file of feature annotations</li>
+ *     </ul>
+ * </p>
+ * <p>
+ *     <h3>Output</h3>
+ *     TSV file of gene expression
+ * </p>
+ *
+ * <p>
+ *     <h3>Usage Examples</h3>
+ *     <p>
+ *         gatk GeneExpressionEvaluation
+ *         -I input.bam
+ *         -G geneAnnotations.gff3
+ *         -O output.tsv
+ *     </p>
+ * </p>
+ * 
+ *<p>Reads are assumed to be paired-end.  Reads which are in a "good pair" are counted once together as a single fragment.  Reads which are not in a "good pair" are each counted separately.
```

Comment:
Consider adding a fragment mate size distance cutoff for this. Document that you currently rely on the aligner

Reply:
Added documentation.  I want to avoid a simple fragment size cutoff since it can be easily thwarted by different protocols or splicing.  And I believe most aligners handle fragment sizes in a more sophisticated manner when setting the properly paired flag.